### ISSUE TITLE
Protect case of proper names

### ISSUE THREAD

**Issue by jedbrown** [citation-file-format/ruby-cff#58]
There does not seem to be a convention for protecting case of proper names so you see
```
title: "XYZ: An ABC alternative"
```
produces the BibTeX
```
title={XYZ: An ABC alternative}
```
which will lose its case when reduced to sentence case by biblatex or pandoc-citeproc. One can hack this for BibTeX (at least as formatted on GitHub citations) using
```
title: "{XYZ}: An {ABC} alternative"
```
but the APA citation and BibTeX key come out mangled. Some convention (could be BibTeX's) should be specified and GitHub's test suite should include this case. (I don't know how closely members of this project might work with the GitHub folks that deployed the sidebar feature, but they don't have a place to file bug reports.)

**Comment by jedbrown** [comment]
I'd suggest it needs to be documented back in the other repo even if the implementation lives in ruby-cff.

**Comment by sdruskat** [comment]
> I'd suggest it needs to be documented back in the other repo even if the implementation lives in ruby-cff.

Hm, I'm not sure. I agree that this should be documented, but not sure the actual CFF schema is the right place for it, as the only client currently (or prospectively) supporting such a syntax is `ruby-cff`... I assume that some of the implementation details for ruby-cff will make it to the GitHub docs page for CITATION.cff files, but don't know how excatly. Do you know more about this @hainesr?

**Comment by hainesr** [comment]
I think it's more a quirk of BibTeX formatting that we forgot to account for in the original implementation, isn't it? For what it's worth I often forget to do the extra `{}` in my BibTeX files first time round 😆 

Anyway, I've added the extra `{}` to the formatter now, so whatever goes into the CFF file should appear as written in the eventual LaTeX output.

**Comment by jedbrown** [comment]
I see you just strip out the `{` and  `}` characters, then wrap the whole title in braces for BibTeX. This isn't sufficient: different bib styles require different capitalization styles (title case versus sentence case). Wrapping the entire source title in braces means the author of the software imposes the case style on the papers of authors who cite them (not okay; they'll have to be edited by hand by every author if they notice). The braces need to be preserved where they are, thus we'd see `title = {Ruby {CFF} Library}` will become "Ruby CFF library" in APA format (uses sentence case). When producing APA format output, the titles should be lowered to sentence case respecting the braced protection.

I also see that titles containing math and other characters (https://github.com/citation-file-format/ruby-cff/commit/0124d1b6238c5e779f8357303fcb9d9694c70f45#diff-b531427c1254019313d13f229b448e9b743ef39f9255e6ca4a7eb8c9835e549cR84) cannot be formatted correctly. Perhaps that's an intentional concession (should be documented if so).

**Comment by jedbrown** [comment]
Ah, and your fix only addresses the BibTeX issue, not the mangled APA output I also reported. Here's an example of mangled APA output.
![image](https://user-images.githubusercontent.com/3303/127522125-cfd5e519-e758-4e87-8f7b-be7831fea4f2.png)

**Comment by jedbrown** [comment]
As for finishing the fix, it seems like incomplete implementations of the schema are not a reason to avoid documenting the schema. And also, it would be valuable for the test suite to be intended for use by all CFF implementations, not just the Ruby implementation.

(Thanks for your work on this. I just think there are more layers to this issue and it's going to be a pain point for lots of people until adequately resolved.)

**Comment by hainesr** [comment]
Sorry, I think we might be at cross purposes here. I looked at your `CITATION.cff` file for the LFAToolkit and you've put `{}`s in there. CFF is more of a meta-format and shouldn't have BibTeX markup in the titles.

If I take your `CITATION.cff` file and remove the `{}`s and run it through (this patched version of) ruby-cff outside of GitHub, I get:

```ruby
puts cff.to_bibtex

@misc{Thompson_LFAToolkit.jl_2021,
author = {Thompson, Jeremy L},
doi = {10.5281/zenodo.5064056},
month = {7},
title = {{LFAToolkit.jl}},
url = {https://jeremylt.github.io/LFAToolkit.jl/stable/},
year = {2021}
}
```

The double `{}`s around the `title` value mean that `LFAToolkit.jl` will be rendered exactly as it's written there. And for APA-like output we get:

```ruby
puts cff.to_apalike

Thompson J.L. (2021). LFAToolkit.jl (version 0.4.0). DOI: https://doi.org/10.5281/zenodo.5064056
```

Aren't they the desired outputs? (Note that GitHub hasn't updated the underlying version of this library yet so you won't see the updated output in the UI yet.)

As for the documentation, it is [documented](https://github.com/citation-file-format/citation-file-format/blob/1.2.0/schema-guide.md), but not fully released yet. GitHub went live with this a couple of weeks earlier than we expected so we're currently running to catch up!

**Comment by jedbrown** [comment]
LFAToolkit was just a handy example. Let's extend my initial example slightly; we want the BibTeX
```
title={{XYZ}: An {ABC} alternative to {libfoo}}
```
so that it can be rendered as "XYZ: An ABC alternative to libfoo" in APA style and as "XYZ: An ABC Alternative to libfoo" in [Chicago style](https://www.chicagomanualofstyle.org/tools_citationguide/citation-guide-1.html#cg-website). This can't be done without escaping. Title case versus sentence case is the decision of the citer (and their journal's requirements), not the author of `CITATION.cff`. The BibTeX way was created for a reason that applies equally here. There should be some way to selectively protect case. It doesn't need to be the BibTeX way, but it will cause ongoing pain and confusion if there is no way. Whatever that way is should be documented.

**Comment by sdruskat** [comment]
Hm, I think I'd like to ideally avoid adding style markup to CFF, although I see why it may not be possible to do so.

I think, however - please correct me if I'm wrong in any of this - that if this concerns the `title` fields only, there are two levels to consider. 

1. The root `title` field describing the software in the repo will only ever contain the "product title" of the software, i.e., even if the `title` of the software is `XYZ: An alternative to libfoo` (or even something natural-language-titled paper-like as `Towards an optimal solution of the XYZ problem as presented in the ABC package via a Python alternative to libfoo` (which would be a highly unusual name for a software)), it won't be covered by title/sentence style considerations, hence can safely be wrapped in `{}` as a whole string to preserve case. This is because the title of the software actually *is* a decision by the authors which should be honoured.
2. The `title` of a reference may well be subject to title/sentence casing and may need markup, which however will not feel unnatural to users.

What do you think? Also patching in @jspaaks to the discussion.

**Comment by jedbrown** [comment]
I don't agree that the entirety of the title field is a proper name. This package has `title: Ruby CFF Library` and could perhaps equally have `title: "ruby-cff: A Ruby library for manipulating CITATION.cff files"`. The problem extends further, however, in that `references` and `preferred-citation` will often have fields that need selective case protection. The blunt hacks will never be able to provide correct bibtex (or APA and Chicago from a single source) -- case protection needs to be a legit feature. There are all sorts of legacy reasons why Google Scholar and other tools have poor case management, but a young standard should do it right so that 20 years from now, we aren't still manually curating to repair case errors in each reference.

**Comment by sdruskat** [comment]
Okay, I think we agree with the need for case protection for `reference` objects :). 

Will repoen and move this back to the main CFF repo where it belongs after all, and happy to continue the discussion there. Which is important to have, as yes, we shouldn't repeat the elders' mistakes :).
